core-kotlin-modules/core-kotlin-exceptions/src/test/kotlin/com/baeldung/conditionalThrowing/ConditionalThrowingUnitTest.kt

```diff
@@ -0,0 +1,117 @@
+package com.baeldung.conditionalThrowing
+
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import kotlin.test.assertEquals
+
+
+class MyException(msg: String) : Exception(msg)
+
+inline fun throwIf(throwCondition: Boolean, exProvider: () -> Exception) {
+    if (throwCondition) throw exProvider()
+}
+
+inline fun <T : Any?> T.mustHave(
+    otherwiseThrow: (T) -> Exception = { IllegalStateException("mustHave check failed: $it") },
+    require: (T) -> Boolean
+): T {
+    if (!require(this)) throw otherwiseThrow(this)
+    return this
+}
+
+data class Player(val id: Int, val name: String, val score: Int)
+class InvalidPlayerException(message: String) : RuntimeException(message)
+
+class ConditionalThrowingUnitTest {
+
+    @Test
+    fun `when using require() then get expected results`() {
+        val str = "a b c"
+        assertThrows<IllegalArgumentException> {
+            require(str.length > 10) { "The string is too short." }
+            null.toString()
+        }
+
+        val upperStr = str.also {
+            require(it.split(" ").size == 3) { "Format not supported" }
+        }.uppercase()
+        assertEquals("A B C", upperStr)
+
+
+        var nullableValue: String? = null
+        assertThrows<IllegalArgumentException> {
+            requireNotNull(nullableValue) { "Null is not allowed" }
+        }
+
+        nullableValue = "a b c"
+        val uppercaseValue = requireNotNull(nullableValue).uppercase()
+        assertEquals("A B C", uppercaseValue)
+    }
+
+    @Test
+    fun `when using check() then get expected results`() {
+        val str = "a b c"
+        assertThrows<IllegalStateException> {
+            check(str.length > 10) { "The string is too short." }
+        }
+
+        var nullableValue: String? = null
+        assertThrows<IllegalStateException> {
+            checkNotNull(nullableValue) { "Null is not allowed" }
+        }
+
+        nullableValue = "a b c"
+        val uppercaseValue = checkNotNull(nullableValue).uppercase()
+        assertEquals("A B C", uppercaseValue)
+    }
+
+
+    @Test
+    fun `when using takeIf() then get expected results`() {
+        val str = "a b c"
+        assertThrows<MyException> {
+            str.takeIf { it.length > 10 } ?: throw MyException("The string is too short.")
+        }
+
+        val nullIsValid: String? = null
+        // we don't expect the exception
```

Comment:
Let's remove this comment too, it's a bit confusing to say in a comment we don't expect the exception then have an assertion showing there is an exception. Or reword the comment to clarify what you mean.

Reply:
✅ removed